recipes/glfw/all/conanfile.py

```diff
@@ -0,0 +1,98 @@
+
+    def system_requirements(self):
+        if self.settings.os == "Linux":
+            package_tool = tools.SystemPackageTool(conanfile=self, default_mode="verify")
+            libs_name = ""
+            os_info = tools.OSInfo()
+            if os_info.with_apt:
+                libs_name = "libx11-dev libxrandr-dev libxinerama-dev libxkbcommon-dev libxcursor-dev " \
+                            "libxi-dev libglu1-mesa-dev"
+            elif os_info.with_yum:
+                libs_name = "libX11-devel libXrandr-devel libXinerama-devel libxkbcommon-devel " \
+                            "libXcursor-devel libXi-devel mesa-libGL-devel"
+            elif os_info.with_pacman:
+                libs_name = "libx11 libxrandr libxinerama libxkbcommon-x11 libxcursor libxi libglvnd "
+            else:
+                self.output.warn("Could not find any package manager, please install x11, xrandr, xinerama," \
+                                 "xkb, xcursor, xi and mesa libraries if not already installed.")
+                return
+            package_tool.install(update=True, packages=libs_name)
+
+    def config_options(self):
+        if self.settings.os == "Windows":
+            del self.options.fPIC
+
+    def configure(self):
+        del self.settings.compiler.libcxx
+        del self.settings.compiler.cppstd
+
+    def source(self):
+        tools.get(**self.conan_data["sources"][self.version])
+        extracted_dir = "{}-{}".format(self.name, self.version)
+        os.rename(extracted_dir, self._source_subfolder)
+
+    def _configure_cmake(self):
+        cmake = CMake(self)
+        cmake.definitions["GLFW_BUILD_EXAMPLES"] = False
+        cmake.definitions["GLFW_BUILD_TESTS"] = False
+        cmake.definitions["GLFW_BUILD_DOCS"] = False
+        cmake.configure(source_folder=self._source_subfolder)
+        return cmake
+
+    def build(self):
+        cmake = self._configure_cmake()
+        cmake.build()
+        if self.settings.os == "Macos" and self.options.shared:
+            with tools.chdir(os.path.join(self._source_subfolder, 'src')):
+                for filename in glob.glob('*.dylib'):
+                    self.run('install_name_tool -id {filename} {filename}'.format(filename=filename))
+
+    def package(self):
+        self.copy("LICENSE.md", dst="licenses", src=self._source_subfolder)
+        self.copy(pattern="*.pdb", dst="bin", keep_path=False)
+        cmake = self._configure_cmake()
+        cmake.install()
+        tools.rmdir(os.path.join(self.package_folder, "lib", "cmake"))
+        tools.rmdir(os.path.join(self.package_folder, "lib", "pkgconfig"))
+
+    def package_info(self):
+        self.cpp_info.libs = tools.collect_libs(self)
+        if self.settings.os == "Linux":
+            self.cpp_info.system_libs.extend(['X11', 'GL', 'm', 'dl', 'pthread'])
+            if self.options.shared:
+                self.cpp_info.exelinkflags.append("-lrt -lm -ldl")
```

Comment:
- Use `system_libs` everywhere.
- The conditional for adding the system libraries should be: `if not self.options.shared`